airflow/providers/docker/operators/docker_swarm.py

```diff
@@ -116,6 +126,8 @@ def __init__(
         networks: list[str | types.NetworkAttachmentConfig] | None = None,
         placement: types.Placement | list[types.Placement] | None = None,
         container_resources: types.Resources | None = None,
+        logging_driver: str | None = None,
```

Comment:
If only `json-file` and `gelf` is supported, maybe we could use `literal` here

Reply:
I think it an ok one as this is more clear.

Reply:
Okay sounds good. Would that error prevent me from pushing? I cannot seem to push the update to the repo.

Reply:
I just tested with a test push and it looks like that error is preventing me from pushing the update. I don't think I can change the type unfortunately.

Reply:
I decided to remove `logging_driver = self.logging_driver.lower()`, which enabled me to add type `Literal`. 

I have also added two unit tests for checking if `logging_driver` works and for entering an invalid/unsupported `logging_driver` value.